src/main/java/mong/Mong.java

```diff
@@ -0,0 +1,358 @@
+    public static void addEvent(String isCompleted, String description, String from, String to) {
+        list.add(new Event(description, from, to));
+        if (isCompleted.contentEquals("0")) {
+            list.get(list.size() - 1).setCompleted(false);
+        } else if (isCompleted.contentEquals("1")) {
+            list.get(list.size() - 1).setCompleted(true);
+        }
+    }
+
+    /**
+     * Add a Task of type Event.
+     * An event consists of the task description and from/to period in String format.
+     */
+    public static void addEvent(String input) {
+        int endOfCommand = input.indexOf("event") + LENGTH_EVENT;
+        int endOfFrom = input.lastIndexOf("/from") + LENGTH_FROM;
+        int startOfFrom = input.indexOf("/from");
+        int endOfTo = input.lastIndexOf("/to") + LENGTH_TO;
+        int startOfTo = input.indexOf("/to");
+        try {
+            String description = input.substring(endOfCommand + 1, startOfFrom - 1);
+            String from = input.substring(endOfFrom + 2, startOfTo - 1);
+            String to = input.substring(endOfTo + 1);
+            list.add(new Event(description, from, to));
+            System.out.println("Mong-ed! This item has been added: ");
+            System.out.println(list.get(list.size() - 1));
+        } catch (StringIndexOutOfBoundsException e) {
+            System.out.println("Oi oi MONG! Missing description, /from or /to commands...");
+        }
+    }
+
+    /**
+     * Delete a Task of any valid type.
+     */
+    public static void deleteTask(String input) {
+        int indexValue = Integer.parseInt(input.split(" ")[1]) - 1;
+        if (indexValue < 0 || indexValue >= list.size()) {
+            throw new NullPointerException();
+        }
+        try {
+            Task deletedTask = list.get(indexValue);
+            list.remove(indexValue);
+            System.out.println("Mong-?!@# This item has been removed: ");
+            System.out.println(deletedTask);
+        } catch (NullPointerException | IndexOutOfBoundsException e) {
+            System.out.println("Mong?!@ Item not in list.");
+        }
+    }
+
+    /**
+     * Returns the TaskType enum value of the first word in the input.
+     */
+    public static TaskType decipherTaskType(String command) throws IllegalTaskTypeException {
+        try {
+            return TaskType.fromCommand(command);
+        } catch (IllegalTaskTypeException e) {
+            throw new IllegalTaskTypeException(e.getMessage());
+        }
+    }
+
+    /**
+     * Adds the command sent by the user into a list.
+     * If the command "list" is sent, the list of previous commands will be printed.
+     * If the command's first word is "mark", it will set the isCompleted variable of the Task to true.
+     * If the command's first word is "unmark", it will set the isCompleted variable of the Task to false.
+     * If the command "bye" is sent, the program will exit.
+     */
+    public static void addByTask() {
```

Comment:
The method addByTask() handles parsing commands and deciding which actions to take, but this could be abstracted further for better readability and separation of concerns.